### ISSUE TITLE
tfe_organization broken in 0.34.0

### ISSUE THREAD

**Issue by lra** [hashicorp/terraform-provider-tfe#570]
#### Terraform Cloud/Enterprise version

Terraform Cloud

#### Terraform version

```plaintext
Terraform v1.2.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/tfe v0.34.0
```

#### Terraform Configuration Files

```terraform
resource "tfe_organization" "SOME_ORG" {
  name                     = "SOME_ORG"
  email                    = "SOME_EMAIL"
  collaborator_auth_policy = "two_factor_mandatory"
}
```

#### Debug Output

A `GET /api/v2/admin/organizations/SOME_ORG` call returns 404.

#### Expected Behavior

```
$ terraform plan
No changes. Your infrastructure matches the configuration.
```

#### Actual Behavior

```
$ terraform plan
╷
│ Error: resource not found
│ 
│   with tfe_organization.seismic,
│   on organization.tf line 1, in resource "tfe_organization" "seismic":
│    1: resource "tfe_organization" "seismic" {
│ 
╵
```

#### Additional Context

We have this simple resource:
```
resource "tfe_organization" "SOME_ORG" {
  name                     = "SOME_ORG"
  email                    = "SOME_EMAIL"
  collaborator_auth_policy = "two_factor_mandatory"
}
```
It was clean up to 0.33.0, but since 0.34.0 we get the above error message.

**Comment by tmatilai** [comment]
Looking at the debug log, a `GET /api/v2/admin/organizations/ICEYE` call returns 404.

The change seems to be #521 (which even put the changelog item into the previous release). Based on PR description the feature only exists in TFE, so maybe it should't be even tried in Terraform Cloud?

**Comment by lra** [comment]
There is no mention of https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/organization being not available to tfcloud, and it was the case before.

Not questioning it, but it should be documented if it's the case, and can we still get the data source https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/data-sources/organization in tfcloud as the org id is needed in a lot of resources?

**Comment by tmatilai** [comment]
No, I meant the admin feature added by #521.

And that indeed is documented:
> * [`admin_settings`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/organization#admin_settings) - Settings that are only available in Terraform Enterprise.

**Comment by lra** [comment]
I see, so this is a regression, will wait for a fix, sticking to `0.33.0` for now.

**Comment by sebasslash** [comment]
:wave: You're right on the money here @tmatilai 

The `tfe_organization` resource should not be calling the Admin Org API, which if your user token does not have the necessary elevated privileges, will result in a 404. 

I'm in favor of removing `admin_settings` from `tfe_organization` entirely and creating a new separate resource `tfe_admin_organization` which manages the Admin Org API. Given this change borked the resource I think it'll be fine patch releasing the removal. 

Looping in @uturunku1 for her thoughts.

**Comment by jmwri** [comment]
This is now working for me again as of `0.35.0`. Thanks for raising and fixing the issue quickly!

**Comment by lra** [comment]
Confirmed: fixed in 0.35.0. Thanks!
